packages/mdc-textfield/label/foundation.js

```diff
@@ -49,32 +49,33 @@ class MDCTextFieldLabelFoundation extends MDCFoundation {
     super(Object.assign(MDCTextFieldLabelFoundation.defaultAdapter, adapter));
   }
 
-  /** Makes the label float above the text field. */
-  floatAbove() {
-    this.adapter_.addClass(cssClasses.LABEL_FLOAT_ABOVE);
-    this.adapter_.removeClass(cssClasses.LABEL_SHAKE);
-  }
-
   /**
-   * Deactivates the label's focus state based on whether the text
-   * field input is empty and passes validity checks.
-   * @param {boolean} shouldRemoveLabelFloat
+   * Styles the label to match the supplied optional params. If a
+   * param is not supplied, the method makes no changes related to that param.
+   * @param {string=} optValue
+   * @param {boolean=} optIsValid
+   * @param {boolean=} optIsBadInput
+   * @param {boolean=} optIsFocused
    */
-  deactivateFocus(shouldRemoveLabelFloat) {
-    this.adapter_.removeClass(cssClasses.LABEL_SHAKE);
+  style(optValue, optIsValid, optIsBadInput, optIsFocused) {
```

Comment:
FYI...I synced your PR up to head, and moved the `styleLabel` logic into just a `style` method on `MDCTextFieldLabelFoundation`. I also removed all the other methods `MDCTextFieldLabelFoundation`, because they no longer seem useful

But! I think this should be broken up into two methods:
* handleInputValueChange(optValue, optIsBadInput, optIsFocused)...this will cause the label to float or not
* handleInputStateChange(optIsValid, optIsFocused)...this will cause the label to shake or not

They seem to be separable, and I always prefer to de-couple things.